### ISSUE TITLE
binutils 2.39 produces bigger arm excutables

### ISSUE THREAD

**Issue by modbw** [crosstool-ng/crosstool-ng#1848]
- crosstool-NG 1.25.0.85_61c4cca
- Ubuntu 20.04 LTS
- arm-unknown-linux-gnueabi sample config

I encountered a problem when building our toolchains and could track it down to the upgrade from binutils 2.38 to 2.39. A simple executable build from `int main(void) {return 0;}` increases in size from about 12k with 2.38 to 72k with 2.39. The only config option I changed in the sample config was binutils 2.39 to binutils 2.38. The output of size is exactly the same for both binaries.

```
~/x-tools-2.39/arm-unknown-linux-gnueabi/bin/arm-unknown-linux-gnueabi-size main-2.39.out 
   text    data     bss     dec     hex filename
   1002     288       4    1294     50e main-2.39.out

 ~/x-tools-2.38/arm-unknown-linux-gnueabi/bin/arm-unknown-linux-gnueabi-size main-2.38.out 
   text    data     bss     dec     hex filename
   1002     288       4    1294     50e main-2.38.out
```

Stripping changes the relation to 5kB to 66kB. I have also attached the objdump -x output of the two executables. Anyone else observed such a behavior?

[objdump-2.38.log](https://github.com/crosstool-ng/crosstool-ng/files/9981336/objdump-2.38.log)
[objdump-2.39.log](https://github.com/crosstool-ng/crosstool-ng/files/9981337/objdump-2.39.log)

**Comment by modbw** [comment]
@dtodor 
Your comment triggered me on this issue and I did some quick research. 

Origin seems to be [this](https://sourceware.org/bugzilla/show_bug.cgi?id=28824) discussion and [this](https://sourceware.org/git/gitweb.cgi?p=binutils-gdb.git;h=9833b7757d246f22db4eb24b8e5db7eb5e05b6d9) commit. Baseline is that binutils changed alignment of relro section from common-page-size (4kB on aarch64) to be max-page-size (64kB) for security reasons.

I don't have a toolchain with binutils >= 2.39 available to test the actual behavior but maybe specifying -z max-page-size=4096 to the linker would solve this issue. We currently use binutils 2.38 until I find some time for testing.

**Comment by dtodor** [comment]
We are building the toolchains ourselves and can test it out.

**Comment by modbw** [comment]
Thanks for testing this. Changing the max-page-size option may have security implications but could fix the overhead produced by binutils >= 2.39. I close the issue as this is not an issue with crosstools-ng.
